src/core/vector/qgsvectorlayer.cpp

```diff
@@ -1266,6 +1266,7 @@ bool QgsVectorLayer::setSubsetString( const QString &subset )
 
   if ( res )
   {
+    mWkbType = mDataProvider->wkbType();
```

Comment:
I think there's a little more required here to support changing geometry types via queries -- eg we'd need to rebuild the layer renderer from a default if the geometry type changes. Could this change be split off into a separate PR?

Reply:
done